spec/models/location_spec.rb

```diff
@@ -26,114 +26,111 @@
   end
 
   describe "#update_location" do
-    let(:district) { FactoryGirl.create(:district) }
-    let(:person)   { FactoryGirl.create(:person, district: district,
-                                                 state:    district.state,
-                                                 zip_code: '03431') }
+    let(:location) { FactoryGirl.create(:location, zip_code: '03431') }
     context "bad address" do
       it "doesn't change location" do
-        expect {
-          person.update_location(address: '2020 Oregon St', zip: 'bad')
-        }.to_not change { person.location }
+        original_attributes = location.attributes
+        location.update_location(address: '2020 Oregon St', zip: 'bad')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.